x/profiles/keeper/alias_functions_test.go

```diff
@@ -265,6 +271,68 @@ func (suite *KeeperTestSuite) TestKeeper_GetApplicationLinks() {
 	suite.Require().Equal(links, suite.k.GetApplicationLinks(ctx))
 }
 
+func (suite *KeeperTestSuite) TestKeeper_IterateExpiringApplicationLinks() {
+	address := "cosmos1y54exmx84cqtasvjnskf9f63djuuj68p7hqf47"
+	links := []types.ApplicationLink{
+		types.NewApplicationLink(
+			address,
+			types.NewData("github", "github-user"),
+			types.ApplicationLinkStateInitialized,
+			types.NewOracleRequest(
+				0,
+				1,
+				types.NewOracleRequestCallData("github", "call_data"),
+				"client_id",
+			),
+			nil,
+			time.Date(2020, 1, 1, 00, 00, 00, 000, time.UTC),
+			time.Date(2022, 1, 1, 00, 00, 00, 000, time.UTC),
+		),
+		types.NewApplicationLink(
+			address,
+			types.NewData("reddit", "reddit-user"),
+			types.ApplicationLinkStateInitialized,
+			types.NewOracleRequest(
+				0,
+				1,
+				types.NewOracleRequestCallData("reddit", "call_data"),
+				"client_id2",
+			),
+			nil,
+			time.Date(2020, 1, 1, 00, 00, 00, 000, time.UTC),
+			time.Date(2022, 1, 1, 00, 00, 00, 000, time.UTC),
+		),
+	}
+
+	// ensure the expiring time is the same of the links
+	suite.ctx = suite.ctx.WithBlockTime(
+		time.Date(2022, 1, 1, 00, 00, 00, 000, time.UTC),
+	)
+	ctx, _ := suite.ctx.CacheContext()
+
+	for _, link := range links {
+		suite.ak.SetAccount(ctx, testutil.ProfileFromAddr(link.User))
+		err := suite.k.SaveApplicationLink(ctx, link)
+		suite.Require().NoError(err)
+	}
+
+	// save an expiration key referring to an already deleted AppLink to test lines 225-227
```

Comment:
It should not include the lines in the reference since it might be changed in the future.